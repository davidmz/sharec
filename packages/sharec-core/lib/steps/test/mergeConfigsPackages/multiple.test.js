const { vol } = require('memfs')
const mergeConfigsPackages = require('../../mergeConfigsPackages')

describe('steps > mergeConfigsPackages > multiple', () => {
  const semaphore = {
    start: jest.fn(),
    success: jest.fn(),
    error: jest.fn(),
    fail: jest.fn(),
  }
  let context

  beforeEach(() => {
    jest.clearAllMocks()
    vol.reset()
  })

  describe('without sharec config', () => {
    it('merges configs sequently', async () => {
      const targetPackage = {
        name: 'awesome-project',
        version: '0.0.0',
        sharec: {
          configs: ['awesome-config', 'next-config'],
        },
      }
      const upcomingConfigs = [
        {
          name: 'awesome-config',
          path: '/node_modules/awesome-config',
          version: '0.0.0',
          configs: {
            '.eslintrc': JSON.stringify({
              env: {
                browser: true,
                es6: true,
              },
            }),
          },
        },
        {
          name: 'next-config',
          path: '/node_modules/next-config',
          version: '0.0.0',
          configs: {
            '.eslintrc': JSON.stringify({
              env: {
                node: true,
                jest: true,
              },
            }),
          },
        },
      ]
      context = {
        targetPath: '/',
        configs: upcomingConfigs,
        mergedConfigs: {},
        options: {},
      }
      const dir = {
        '/package.json': JSON.stringify(targetPackage),
      }
      vol.fromJSON(dir, '/configs')

      const output = await mergeConfigsPackages(context, semaphore)

      expect(output.mergedConfigs['/.eslintrc']).toMatchSnapshot()
    })
  })
})
