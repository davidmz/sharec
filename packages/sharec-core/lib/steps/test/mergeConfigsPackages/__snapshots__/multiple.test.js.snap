// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`steps > mergeConfigsPackages > multiple without sharec config merges configs sequently 1`] = `
"{
  \\"env\\": {
    \\"browser\\": true,
    \\"es6\\": true,
    \\"node\\": true,
    \\"jest\\": true
  }
}"
`;
